answerking_app/utils/mixins/GenericMixins.py

```diff
@@ -0,0 +1,34 @@
+from MySQLdb.constants.ER import DUP_ENTRY
+from django.db import IntegrityError
+from rest_framework import status
+from rest_framework.mixins import CreateModelMixin, UpdateModelMixin
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+
+class CreateMixin(CreateModelMixin):
+    def create(self, request: Request, *args, **kwargs) -> Response:
+        try:
+            return super().create(request, *args, **kwargs)
+        except IntegrityError as exc:
+            if exc.args[0] == DUP_ENTRY:
```

Comment:
a bit ugly that the code is repeating, maybe move the if/else part in a function